bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/discovery/SomfyTahomaItemDiscoveryService.java

```diff
@@ -455,17 +458,18 @@ private void deviceDiscovered(String label, String deviceURL, String oid, ThingT
 
         SomfyTahomaBridgeHandler localBridgeHandler = bridgeHandler;
         if (localBridgeHandler != null) {
-            ThingUID thingUID = new ThingUID(thingTypeUID, localBridgeHandler.getThing().getUID(), oid);
+            ThingUID thingUID = new ThingUID(thingTypeUID, localBridgeHandler.getThing().getUID(),
+                    deviceURL.replaceAll("[\\W]|_", ""));
```

Comment:
Please rather use `replaceAll("[^a-zA-Z0-9_]", "")` to be sure  having a valid thing ID.

Reply:
OK, done